lib/router/index.js

```diff
@@ -59,6 +61,24 @@ function throwIfNoRouter () {
 // Export the SingletonRouter and this is the public API.
 export default SingletonRouter
 
+export function withRouter (options) {
+  return function withRouterComponent (Comp) {
+    const compName = Comp.displayName || Comp.name || 'Component'
+
+    return class ComponentWithRouter extends Component {
+      static contextTypes = {
+        router: PropTypes.object
+      }
+
+      static displayName = `withRouter(${compName})`
+
+      render () {
+        return <Comp {...this.props} router={this.context.router} />
+      }
+    }
+  }
+}
+
```

Comment:
This should be its own file so you can exclude it from the the build if you dont use it

#codeSplitting